### ISSUE TITLE
`shipit production db:push` does not work

### ISSUE THREAD

**Issue by gridsystem** [timkelty/shipit-db#24]
There seem to be several issues with the code causing this, one of which is that `shipit.config.default.db` is overwritten with `shipit.config.production.db`, meaning the local credentials stored in `shipit.config.default.db.local` are overwritten.

Has it always been this way?

**Comment by timkelty** [comment]
@gridsystem Really, sorry – I haven't used Shipit or done any work on it in years, so support is tough.

However! I did a little digging and it appears the last version I had this working on was `shipit-cli@1.5.2`, which does a merge instead of assign on the environment configs, allowing this to work.

Apparently that changed in 2.0.

You could always just _not_ use  `shipit.config.default.db`, and set both `shipit.config.production.db.local` and `shipit.config.production.db.remote`…I think?

Again, very sorry I'm unable to support better.

**Comment by gridsystem** [comment]
Hey @timkelty,

Thanks for your response! I actually figured that would work and managed to set it up exactly the way you described.

Out of curiosity what tools set are you using in place of shipit?

**Comment by timkelty** [comment]
@gridsystem Couldn't be more different :)

We're using a CI/CD approach (we use [Gitlab](https://about.gitlab.com/product/continuous-integration/)) now, along with Docker, so when we push to our repos, it triggers a build on a remote server and rebuilds the whole application.

Prior to that, we we went back to using Capistrano. Shipit was initially very appealing because it was JS based and we wanted to rid our projects of the Ruby dependency.

However, after using it and working on it for a long time, we found deployments to be just too critical to leave to a project like this (not a huge user base, not extensively "battle tested").

…Not to say it can't work for you, it just ended up costing more time that we wanted to put into it.

I know @neoziro has been pushing lots to the newer versions, so it could be worth another look.
